weed/s3api/auth_credentials.go

```diff
@@ -381,8 +400,18 @@ func (iam *IdentityAccessManagement) authRequest(r *http.Request, action Action)
 	}
 
 	glog.V(3).Infof("user name: %v actions: %v, action: %v", identity.Name, identity.Actions, action)
-
+	glog.V(3).Infof("request: %v", r)
 	bucket, object := s3_constants.GetBucketAndObject(r)
+	prefix := s3_constants.GetPrefix(r)
+
+	glog.V(3).Infof("bucket: %v, object: %v, prefix: %v", bucket, object, prefix)
+
+	// If object is empty, and the prefix is not empty, then the object is the prefix
+	if object == "/" && prefix != "" {
+		object = prefix
+	}
```

Comment:
When using any of these:
- `aws s3 ls s3://<BUCKET_NAME>/<DIRECTORY>/*'`
- `aws s3api list-objects-v2 --bucket <BUCKET_NAME> --prefix <DIRECTORY> --delimiter '/'
- `boto3.client(...).list_objects_v2(Bucket=bucket_name, Prefix=path)`

The object returned from `GetBucketAndObject` was always `/`.
The query string `prefix` though did include the directory, so now when we can't get the Object from `GetBucketAndObject`, and there is a prefix available, we use that prefix instead.